packages/bpk-component-map/src/BpkMap.js

```diff
@@ -18,11 +18,133 @@
 
 /* @flow */
 
-import React from 'react';
+import React, { type Node } from 'react';
+import PropTypes from 'prop-types';
 import { withGoogleMap, GoogleMap } from 'react-google-maps';
+import { LatLongPropType, type LatLong } from './common-types';
 
-const BpkMap = withGoogleMap(({ mapRef, ...rest }) => (
-  <GoogleMap ref={mapRef} {...rest} />
-));
+export type Bounds = {
+  south: number,
+  west: number,
+  north: number,
+  east: number,
+};
+
+export type MapRef = ?{
+  getBounds: () => Bounds,
+  getCenter: () => LatLong,
+  getZoom: () => number,
+  fitBounds: Bounds => void,
+};
+
+type Props = {
+  scrollEnabled: boolean,
+  zoom: number,
+  zoomEnabled: boolean,
+  bounds: ?Bounds,
+  center: ?LatLong,
+  children: ?Node,
+  containerElement: ?Node,
+  mapElement: ?Node,
+  mapRef: ?(MapRef) => mixed,
+  onRegionChange: ?(Bounds, LatLong) => mixed,
+  onZoom: ?(number) => mixed,
+};
+
+const BpkMap = withGoogleMap((props: Props) => {
+  const {
+    bounds,
+    children,
+    mapRef,
+    onRegionChange,
+    onZoom,
+    center,
+    scrollEnabled,
+    zoom,
+    zoomEnabled,
+  } = props;
+  let ref: MapRef = null;
+
+  if (!bounds && !center) {
+    throw new Error('BpkGoogleMap: Provide either `bounds` or `center` props.');
+  }
+
+  return (
+    <GoogleMap
+      ref={(map: MapRef) => {
+        ref = map;
+        if (map && bounds) {
+          map.fitBounds({
+            south: bounds.south,
+            west: bounds.west,
+            north: bounds.north,
+            east: bounds.east,
+          });
+        }
+        if (mapRef) {
+          mapRef(map);
+        }
+      }}
+      defaultCenter={
+        center
+          ? {
+              lat: center.latitude,
+              lng: center.longitude,
+            }
+          : null
+      }
+      defaultZoom={zoom}
+      options={{
+        gestureHandling: scrollEnabled ? 'auto' : 'none',
+        zoomControl: zoomEnabled,
```

Comment:
We would need to be able to enable/disable more controls: `draggable`, `mapTypeControl`, `streetViewControl`, `fullscreenControl`, `rotateControl`.

This is the wrapper we use around the current `BpkMap`, and the options we set: https://gitlab.skyscannertools.net/hotels-web/hotels-website/blob/master/client/src/components/MapWrapper/MapWrapper.jsx#L132-140

Also, what do you think about adding a `dynamic` prop that when set to false disables all controls and interactions? This is useful when using the map as a static map (in our case, to show an "image" of the hotel location).

You can see our current map usage, both static and dynamic here: http://hotels-website.slingshot.eu-west-1.sandbox.aws.skyscanner.local/47016457

Reply:
Thanks for this. I'll update it to support this. The links are very helpful too.

> Also, what do you think about adding a dynamic prop that when set to false disables all controls and interactions? This is useful when using the map as a static map (in our case, to show an "image" of the hotel location).

I like the intention of this, but I think we should hold off on it for now. The reason I say this is that we may want to achieve the same behaviour in a different way, for example a component that displays a static map image (so it would look the same, but actually just show an image), which may help us use fewer API calls. We're talking to Gadget squad about doing something like this in the future.

Reply:
Great! It would be very good to be able to use just an image when showing an static map :-)

Keep us updated on that component, as you can see we'll use it also :-P
For now we'll keep disabling all interactions and controls when "simulating" the static map.

Reply:
`draggable` is [deprecated](https://developers.google.com/maps/documentation/javascript/reference/3.exp/map#MapOptions.draggable), they recommend using `gestureHandling: 'none'`, which we're doing in this file on line 98.

Reply:
I think we can use [`disableDefaultUI`](https://developers.google.com/maps/documentation/javascript/reference/3.exp/map#MapOptions.disableDefaultUI) as a way to toggle all of those controls at once (except draggable). What do you think about us using that instead - can you foresee any issues on your end? For example, a situation where you'd want to disable some but not others.

Reply:
Good one about `draggable` :-D

About `disableDefaultUI`, as we discussed (and just for the record here), we need to be able to disable just some controls, like full screen, satellite view, streetview... so we'll still require fine grained access to the different controls.